### PR TITLE
Allow pokemon to evolve again after breeding, apply custom evolutions, give tyrogue, eevee cutom evolution

### PR DESCRIPTION
**Pull request by RedSparr0w**
Refer #363 

This allows Tyrougue to evolve correctly, also allows you to obtain espeon/umbreon.
And can also be used for other pokemon in the future (Happiness based etc..)

Current problems that will need to be addressed at some point:
-  ~~The pokemon can only evolve once as the `evolved` flag is not reset on when breeding~~
- ~~Even if you reset the `evolved` flag once they hatch, the evolver observer isn't started again~~

### CODE REVIEW

**Comment by fuws** [comment]
So with this, the player may still fail to get Hitmontop during the prestige?

**Comment by RedSparr0w** [comment]
Yes, but in a seperate PR I will update the breeding functions to reset the caught pokemon `evolved` state to `false`, and re attach the evolver observer.

So you may have to breed Tyrogue a bunch of times before getting Hitmontop

**Comment by RedSparr0w** [comment]
Eh, decided to fix it in this PR.

Now any pokemon can evolve again after hatching.

On a side note, it looks like it is impossible to obtain a shiny pokemon from evolving.

File: src/scripts/pokemons/CaughtPokemon.ts [hunk 1]
```diff
@@ -49,4 +38,38 @@ class CaughtPokemon {
         plainJS = ko.toJS(this);
         return Save.filter(plainJS, keep);
     }
+
+    public evoTrack(reset = false){
```
Comment (RedSparr0w) [inline src/scripts/pokemons/CaughtPokemon.ts]: I don't know what to call this, taking any and all suggestions!

**Comment by RedSparr0w** [comment]
I've added another custom function for eevee,
But it might work better as an array?

```javascript
        "evolution": ["Vaporeon",    "Jolteon",       "Flareon",    ["Espeon", "Umbreon"]],
        "evoLevel":  ["Water_stone", "Thunder_stone", "Fire_stone", "Time_stone"],
```

**Comment by RedSparr0w** [comment]
>I think I'm missing something, how is it decided if you get Hitmontop?

It's decided randomly

https://github.com/Ishadijcks/pokeclicker/pull/368/files#diff-3e703bba7e316c46a04035db1cfbba4eR3174

File: src/scripts/pokemons/CaughtPokemon.ts (hunk 1 above)
Comment (Ishadijcks) [reply src/scripts/pokemons/CaughtPokemon.ts]: `checkForEvolution()`?

**Comment by Ishadijcks** [comment]
Can you fix the merge conflicts please?

**Comment by RedSparr0w** [comment]
Conflicts resolved 👍

**Pull request closed without merging**